src/cosmicqc/analyze.py

```diff
@@ -175,26 +178,29 @@ def find_outliers(
             Outlier data frame for the given conditions.
     """
 
-    # interpret the df as CytoDataFrame
-    df = CytoDataFrame(data=df)
-
+    # Resolve feature_thresholds if provided as a string
     if isinstance(feature_thresholds, str):
         feature_thresholds = read_thresholds_set_from_file(
             feature_thresholds=feature_thresholds,
             feature_thresholds_file=feature_thresholds_file,
         )
 
-    # Filter DataFrame for outliers using all conditions
-    outliers_df = df[
-        # use identify outliers as a mask on the full dataframe
-        identify_outliers(
-            df=df,
-            feature_thresholds=feature_thresholds,
-            feature_thresholds_file=feature_thresholds_file,
-        )
-    ]
+    # Determine the columns required for processing
+    required_columns = list(feature_thresholds.keys()) + metadata_columns
 
-    # Print outliers count and range for each feature
+    # Interpret the df as CytoDataFrame
+    df = CytoDataFrame(data=df)[required_columns]
```

Comment:
Is this a default behavior? I like the idea of improving speed here by not needing all the columns!

Reply:
This is a great question! I'll reassess here to ensure one can still read all the columns if necessary.

Reply:
Opened #152